### DISCUSSION TITLE
[SOLVED] Today I broke the `main` branch 😫

### DISCUSSION THREAD

**Discussion by bglamadrid** [trebol-ecommerce/trebol-backend-monolith#173] [Announcements]
So please do not pull changes from it and wait for me to fix it...

EDIT: I managed to restore its history. Still, one piece of advice: if you were working on top of its most recent state (24 hours ago), **you should backup your work** AND fetch and pull latest changes ASAP. Then, if merging it breaks your work branch, at least you'll be able to re-create that branch again.

Sorry for the disruption.

**Comment by mepox** [comment]
Yeah I just noticed it 😄

**Comment by mepox** [comment]
@bglamadrid I have the old history in my fork if that helps?

Edit: I mean I was about to open a PR #174 and lots of commits showed up. I guess those are missing from the main.

**Reply by bglamadrid** [reply]
Yeah, actually I am making use of your fork history to get this one's back 🙂

My big issue here has to do with merge commits and the implicit merge operation done by `git pull`...

See, we devs never stop (failing miserably and) learning!

**Reply by bglamadrid** [reply]
Alright, now everything's back to normal on the remote I think... Still, merging might be a pain.

I'll reopen your PR and we'll see how that goes.

**Reply by bglamadrid** [reply]
From what I can see, even if I did recover the original history, the commits are not the same. The PR still wants to add what's already in the codebase. It won't do...

My suggestion is that you recreate the branch and the PR based on this latest rollup

- Backup the `ProductListJpaCrudServiceImpl.java` file
- Switch to your `main` branch (pulling latest changes)
- Remove your `issue-171` branch (on your local repo, not on your fork remote)
- Create a new local branch off the latest `main`, using the exact same name `issue-171` if you wish
- Add & commit the `ProductListJpaCrudServiceImpl.java` file and all notable changes, to this new branch
- Force-push changes from this new branch to your remote `issue-171` branch in your fork
- Create a new PR

Sorry for the mess I made. I will do my best to never let this happen again.

**Reply by mepox** [reply]
No worries. I already backed it up. 😄 I will re-create the PR once it's ready.
